### ISSUE TITLE
Unrecognised item type error when using fields on GetPlaylistItems

### ISSUE THREAD

**Issue by thechubbypanda** [zmb3/spotify#262]
I'm calling 
```
spotify.Client.GetPlaylistItems(ctx, id, spotify.Fields("items(track(id))"))
```

and I get the error:

```
unrecognized item type: 
```

I'm assuming this is because it's expecting to decode the rest of the fields that don't exist.

I'd expect it to just give me back tracks with only the id populated.

**Comment by TimHi** [comment]
I'm confused what `spotify.Fields("items(track(id))")` is in your case, are you passing a valid spotify playlist ID?

**Comment by thechubbypanda** [comment]
Yeah, the playlistID is valid. The call works fine without the Fields parameter. 
See the code: https://github.com/thechubbypanda/syncify/blob/0cf5e36332d9c4fac0dfe627b9fb238bdaab9026/sync.go#L81

**Comment by TimHi** [comment]
I see, the unmarshaling of the JSON expects the item type, using the filter its missing in `func (t *PlaylistItemTrack) UnmarshalJSON`. 

Also it doesnt work for filters like `spotify.Fields("items(track(name))")`. 

I guess when using a Field it can't be expected that the `PlaylistItemTrack` matches the received JSON.

**Comment by thechubbypanda** [comment]
So is there something we can do about this or is it a lost cause?
Do other endpoints play nicely with the fields param?
